test/test_util.py

```diff
@@ -256,6 +258,39 @@ def test_make_headers(self):
             make_headers(disable_cache=True),
             {'cache-control': 'no-cache'})
 
+    def test_rewind_body(self):
+        body = io.BytesIO(b'test data')
+        self.assertEqual(body.read(), b'test data')
+
+        # Assert the file object has been consumed
+        self.assertEqual(body.read(), b'')
+
+        # Rewind it back to just be b'data'
+        rewind_body(body, 5)
+        self.assertEqual(body.read(), b'data')
+
+    def test_rewind_body_failed_tell(self):
+        body = io.BytesIO(b'test data')
+        body.read()  # Consume body
+
+        # Simulate failed tell()
+        self.assertRaises(UnrewindableBodyError, rewind_body, body, object())
+
+    def test_rewind_body_bad_position(self):
+        body = io.BytesIO(b'test data')
+        body.read()  # Consume body
+
+        # Pass non-integer position
+        self.assertRaises(UnrewindableBodyError, rewind_body, body, None)
```

Comment:
I think I originally entertained that idea with Requests implementation, but then we made `_body_position` "private". The common case we're catching here with an invalid position is the `_FAILEDTELL` sentinel object, which the user isn't setting. It made more sense to hide that abstraction from people not looking to go digging through the code for what `ValueError: object() is not a valid file position.`

It may make more sense here since this is a public parameter to generalize the error, I don't have strong feelings either way. This was just legacy logic from my original approach.